src/App.tsx

```diff
@@ -1,27 +1,73 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import { Button, ButtonGroup } from '@mui/material';
 import { GoodsList } from './GoodsList';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleClickAllGoods = async () => {
+    try {
+      const goods = await getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      setGoodsList(goods);
+    } catch (error) {
+      throw new Error('No data was found!');
+    }
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleClickFirstFiveGoods = async () => {
+    try {
+      const goods = await get5First();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      setGoodsList(goods);
+    } catch (error) {
+      throw new Error('No data was found!');
+    }
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleClickRedGoods  = async () => {
+    try {
+      const goods = await getRedGoods();
+
+      setGoodsList(goods);
+    } catch (error) {
+      throw new Error('No data was found!');
+    }
```

Comment:
all what you change in your functions is API call so let's create separate fucntion for `try...cath` and `setGoodsList`

Reply:
fixed